admin-password-updater/updater/event_handler.go

```diff
@@ -0,0 +1,172 @@
+	for {
+		select {
+		case event, ok := <-u.Watcher.Events:
+			if !ok {
+				u.Log.V(0).Info("watcher events channel is closed, exiting...", "directory", u.WatchDir)
+				u.Done <- true
+				return
+			}
+			if fileChanged(u.DefaultUserFile, event) {
+				u.Log.V(2).Info("file system event", "file", u.DefaultUserFile, "operation", event.Op.String())
+
+				// read default user username and (new) password
+				defaultUserCfg, err := ini.Load(u.DefaultUserFile)
+				if err != nil {
+					u.Log.Error(err, "failed to load INI data source", "file", u.DefaultUserFile)
+					u.Done <- true
+					return
+				}
+				defaultUser := defaultUserCfg.Section("").Key("default_user").String()
+				newPasswd := defaultUserCfg.Section("").Key("default_pass").String()
+
+				// read admin username and (old) password
+				adminCfg, err := ini.Load(u.AdminFile)
+				if err != nil {
+					u.Log.Error(err, "failed to load INI data source", "file", u.AdminFile)
+					u.Done <- true
+					return
+				}
+				adminSection := adminCfg.Section("default")
+				adminUser := adminSection.Key("username").String()
+				oldPasswd := adminSection.Key("password").String()
+
+				if defaultUser != adminUser {
+					u.Log.V(0).Info("exiting because usernames do not match",
+						"default-user", defaultUser, "default-user-file", u.DefaultUserFile,
+						"admin-user", adminUser, "admin-file", u.AdminFile)
+					u.Done <- true
+					return
+				}
+				if newPasswd == oldPasswd {
+					u.Log.V(2).Info("passwords already match, nothing to do", "username", defaultUser)
+					break
+				}
+
+				u.Rmqc.SetUsername(adminUser)
+				u.Rmqc.SetPassword(oldPasswd)
+
+				if !u.updateInRabbitMQ(adminUser, newPasswd) {
+					break
+				}
+
+				u.Log.V(4).Info("copying new password...", "source", u.DefaultUserFile, "target", u.AdminFile)
+				adminSection.Key("password").SetValue(newPasswd)
+				if err := adminCfg.SaveTo(u.AdminFile); err != nil {
+					u.Log.Error(err, "failed to write new password", "file", u.AdminFile)
+					u.Done <- true
+					return
+				}
+				u.Log.V(2).Info("copied new password", "source", u.DefaultUserFile, "target", u.AdminFile)
+			} else {
+				u.Log.V(4).Info("file system event", "file", event.Name, "operation", event.Op.String())
+			}
+
+		case err, ok := <-u.Watcher.Errors:
+			if !ok {
+				u.Log.V(0).Info("watcher errors channel is closed, exiting...")
+				u.Done <- true
+				return
+			}
+			u.Log.Error(err, "failed to watch", "directory", u.WatchDir)
+		}
+	}
+
+}
+
+func fileChanged(defaultUserFile string, event fsnotify.Event) bool {
+	return event.Name == defaultUserFile &&
+		(event.Op&fsnotify.Create == fsnotify.Create ||
+			event.Op&fsnotify.Write == fsnotify.Write)
+}
+
+// updateInRabbitMQ sets newPasswd for existingUser in the RabbitMQ server.
+// It returns true if password could be sucessfully updated or password was already up-to-date.
+func (u *PasswordUpdater) updateInRabbitMQ(existingUser, newPasswd string) (ok bool) {
```

Comment:
The function signature here I found rather confusing. The functions nested within this one all refer to failures, error handling, etc., which make me think this should return an error rather than a bool

Reply:
It now returns an error. Thank you.